js/researches/german/getSentenceParts.js

```diff
@@ -0,0 +1,71 @@
+var stopwords = require( "./passivevoice-german/stopwords.js" )();
+var arrayToRegex = require( "../../stringProcessing/createRegexFromArray.js" );
+var stripSpaces = require( "../../stringProcessing/stripSpaces.js" );
+var SentencePart = require( "./GermanSentencePart.js" );
+var auxiliaries = require( "./passivevoice-german/auxiliaries.js" )();
+
+var forEach = require( "lodash/forEach" );
+var isEmpty = require( "lodash/isEmpty" );
+var map = require( "lodash/map" );
+
+var stopwordRegex = arrayToRegex( stopwords );
+var auxiliaryRegex = arrayToRegex( auxiliaries );
+
+/**
+ * Strips spaces from the auxiliary matches.
+ *
+ * @param {Array} matches A list with matches of auxiliaries.
+ * @returns {Array} A list with matches with spaces removed.
+ */
+function sanitizeMatches( matches ) {
+	return map( matches, function( match ) {
+		return stripSpaces( match );
+	} );
+}
+
+/**
+ * Splits sentences into sentence parts based on stopwords.
+ *
+ * @param {string} sentence The sentence to split.
+ * @param {Array} stopwords The array with matched stopwords.
+ * @returns {Array} The array with sentence parts.
+ */
+function splitOnWord( sentence, stopwords ) {
+	var currentSentence = sentence;
+	var sentenceParts = [];
+	forEach( stopwords, function( stopword ) {
+		var splitSentence = currentSentence.split( stopword );
```

Comment:
Fixed this by first splitting sentences and creating sentence part of that array.